dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/datastore/DefaultDatastoreService.java

```diff
@@ -190,6 +211,23 @@ private <T> T readProtectedIn(String namespace, T whenHidden, Supplier<T> read)
     return whenHidden;
   }
 
+  private <T> boolean isDatastoreEntryWithNoProtection(
+      T res, DatastoreNamespaceProtection protection) {
+    return (res instanceof DatastoreEntry) && (protection == null);
+  }
+
+  private <T> boolean isDatastoreEntryAndProtectionSharingRespected(
+      T res, DatastoreNamespaceProtection protection) {
+    return (res instanceof DatastoreEntry)
+        && (protection != null && protection.isSharingRespected());
```

Comment:
I wonder if we simply always do ACL now. When only protected namespaces had sharing as an option it made sense to check this but now if all entries are by default sharing protected it does not make much sense that a protected namespace basically can opt-out of the sharing level protection.

Reply:
happy to update if that's what we want to do from now on.

Reply:
I think we should :)

Reply:
ok, so we'll get rid of the `sharingRespected` property as sharing will always apply from now on. I'll work on that in a separate issue then to keep things separate as it impacts writes as well as reads